src/Symfony/Component/JsonPath/JsonPathUtils.php

```diff
@@ -85,4 +85,96 @@ public static function findSmallestDeserializableStringAndPath(array $tokens, mi
             'tokens' => $remainingTokens,
         ];
     }
+
+    public static function unescapeString(string $str): string
+    {
+        $result = '';
+        $length = strlen($str);
+
+        for ($i = 0; $i < $length; $i++) {
+            if ('\\' === $str[$i] && $i + 1 < $length) {
+                $nextChar = $str[$i + 1];
+
+                switch ($nextChar) {
+                    case '"':
+                        $result .= '"';
+                        $i++;
+                        break;
+                    case "'":
+                        $result .= "'";
+                        $i++;
+                        break;
+                    case '\\':
+                        $result .= '\\';
+                        $i++;
+                        break;
+                    case '/':
+                        $result .= '/';
+                        $i++;
+                        break;
+                    case 'b':
+                        $result .= "\b";
+                        $i++;
+                        break;
+                    case 'f':
+                        $result .= "\f";
+                        $i++;
+                        break;
+                    case 'n':
+                        $result .= "\n";
+                        $i++;
+                        break;
+                    case 'r':
+                        $result .= "\r";
+                        $i++;
+                        break;
+                    case 't':
+                        $result .= "\t";
+                        $i++;
+                        break;
+                    case 'u':
+                        if ($i + 5 < $length) {
```

Comment:
nesting is deep here, I'm wondering if we couldn't use a match instead of a switch, and the `u` case, call a method instead, which would use early returns? `$i` passed by ref of course.

Reply:
Good idea, looks way better now